### ISSUE TITLE
Provide UI to control whether thumbnails have text superimposed

### ISSUE THREAD

**Issue by DavidPhillipOster** [DavidPhillipOster/ThumbHost3mf#6]
… would it be possible to turn off the green 3MF and red GCODE text? Could be a setting in preferences if some people like having the text. I know if I clicked on a .3MF or a .gcode file and I would rather not have the text cover up the object I want to see. Thank you for your consideration. I feel the same way about the folded-over corner, too.

_Originally posted by @alsmithson in https://github.com/DavidPhillipOster/ThumbHost3mf/issues/5#issuecomment-1926156246_

**Comment by DavidPhillipOster** [comment]
The 3MF host app could certainly have a Settings dialog box that wrote to a Preferences file. I think that the plugin could then see it.

The folded corner is the default behavior of the operating system. I'd have to research whether it can be turned off.

If the option existed, I'd have to research if there is an API to get Quicklook to re-scan files to get the current icon.

The reason the labeling behavior is on by default is to support Mac users who have kept the default setting of hiding file extensions.

Note that you can give files custom icons by:

Position the camera to the best angle, use ⌘⇧4 then the mouse to sweep out a rectangle for a screenshot, open the screenshot, and do a select all, copy, then ⌘i on the file to open a Get Info window, use the mouse to click on the icon in the upper left of the window, and paste. It sounds tedious written out in text, but it is only a few seconds in practice.

i.e.:

$OACIV - all with the command key held down.

The $OACIV trick is quite fast. I added icons to a dozen files as an experiment.

The custom icon overrides the ThumbHost3mf icon, but if you copy a file to a FAT-32 SD card, I'm under the impression that the custom icon is not preserved.

**Comment by alsmithson** [comment]
Thanks for your detailed reply. I never thought about people who hide extensions as I always change that setting to show them. They would surely benefit from the text on the icon. Now I do see a couple files that have the folded corner, but it is pretty rare on my system. XLSX files use them while DMG files use a fancier curled corner. Py and md files also have the folded corner. All of the ones I've found are white on white, so I never really noticed the corner. I don't care for it, but I can certainly live with it. 

Thanks for the reminder about custom icons. I used that in the past, but forgot how. I recently searched, unsuccessfully, how to change folder icons. I tried this just now and I see that it works on folders too!

**Comment by MiroIsNot1337** [comment]
I know you have your reasons for adding the text, but I'd also second disabling that by default or at least making the text a lot smaller and not position it in the center. I think users that leave the file extensions hidden will probably not care about this filetype either.

**Comment by DavidPhillipOster** [comment]
Version 1.5 adds a setup dialog box to let you control whether quicklook adds a text label with the filetype or not. Because generated icons are cached, it may some time before you see the result of the change. To test this, I used a set of MS-DOS (FAT32) formatted thumb drives containing .gcode files that I knew had thumbnails. I closed the ThumbHost3mf app and restarted the Mac in between changing a setting a restarting. 
The plugin and the app communicate using a shared app group, documented here: https://developer.apple.com/documentation/xcode/configuring-app-groups
